tiledb/sm/filesystem/mem_filesystem.cc

```diff
@@ -43,82 +46,167 @@ namespace sm {
 
 class MemFilesystem::FSNode {
  public:
-  FSNode(const std::string& name, FSNode* parent)
+  /* ********************************* */
+  /*     CONSTRUCTORS & DESTRUCTORS    */
+  /* ********************************* */
+
+  /** Value constructor. */
+  FSNode(const std::string& name, FSNode* const parent)
       : name_(name)
-      , parent_(parent){};
+      , parent_(parent) {
+  }
+
+  /** Copy constructor. */
+  DISABLE_COPY(FSNode);
 
+  /** Move constructor. */
+  DISABLE_MOVE(FSNode);
+
+  /** Destructor. */
   virtual ~FSNode() = default;
 
-  /* Indicates if it's a file or a directory */
+  /* ********************************* */
+  /*             OPERATORS             */
+  /* ********************************* */
+
+  /** Copy-assignment. */
+  DISABLE_COPY_ASSIGN(FSNode);
+
+  /** Move-assignment. */
+  DISABLE_MOVE_ASSIGN(FSNode);
+
+  /* ********************************* */
+  /*                API                */
+  /* ********************************* */
+
+  /** Indicates if it's a file or a directory */
```

Comment:
@ypatia I forgot to mention this earlier: the `/**` is the syntax that our in-code doc generator uses. Use this for public block comments.